core/src/main/java/nu/yona/server/subscriptions/service/NewDeviceRequestDTO.java

```diff
@@ -0,0 +1,48 @@
+package nu.yona.server.subscriptions.service;
+
+import java.util.Date;
+
+import nu.yona.server.subscriptions.entities.NewDeviceRequest;
+
+import com.fasterxml.jackson.annotation.JsonRootName;
+
+@JsonRootName("newDeviceRequest")
+public class NewDeviceRequestDTO
+{
+	private Date expirationDateTime;
+	private String userPassword;
+
+	public NewDeviceRequestDTO(Date expirationDateTime, String userPassword)
+	{
+		this.expirationDateTime = expirationDateTime;
+		this.userPassword = userPassword;
+	}
+
+	public Date getExpirationDateTime()
+	{
+		return expirationDateTime;
+	}
+
+	public String getUserPassword()
+	{
+		return userPassword;
+	}
+
+	public static NewDeviceRequestDTO createInstance(NewDeviceRequest newDeviceRequestEntity)
+	{
+		if (newDeviceRequestEntity == null)
```

Comment:
Reason was that the client can do a valid call to check if there are any live requests to display it on the app to the user. Now throwing exception and returning 404 (the client can use that too).